crates/cairo-profiler/src/trace_reader/function_trace_builder/function_stack_trace.rs

```diff
@@ -1,7 +1,7 @@
+use crate::trace_reader::function_name::FunctionName;
 use crate::trace_reader::function_trace_builder::Steps;
-use crate::trace_reader::functions::FunctionName;
 
-pub(super) struct Function {
+pub(super) struct FunctionOnStack {
```

Comment:
?
this is weird, you do not name the int type `IntInVector`

Reply:
This struct only makes sense in context of the stack (recursive function calls, steps in the moment of the call etc.), there is other struct names `Function` where the name is more suitable. I can rename to `FunctionStackElement` if that's better, but all the fields are related to one function on the stack, that's why I chose the name

Reply:
> This struct only makes sense in context of the stack (recursive function calls, steps in the moment of the call etc.),

I believe the, but it still is a bad name. I can construct it as a standalone type and it will not be on stack.

> `FunctionStackElement` 

Imo still bad. This is just bad abstraction. Compression is an impl detail of the stack. you can name it like this but don't make it public.

Reply:
Yea, it actually doesn't need to be public, oversight from my side